src/cmd/ksh93/sh/fault.c

```diff
@@ -119,6 +156,10 @@ void	sh_fault(int sig)
 	char		*trap;
 	struct checkpt	*pp = (struct checkpt*)shp->jmplist;
 	int	action=0;
+
+	if(sig==SIGSEGV) {
+	    dump_backtrace(100, 0);
```

Comment:
I see that you have installed this function as a signal handler. Why we need to call it again here ?

Reply:
This ensures we get a backtrace if the user has done `trap 'do_something' SEGV`. Personally I think ksh should disallow trapping a handful of signals including SIGSEGV. There is absolutely nothing ksh script can meaningfully do on receipt of that signal. SIGSEGV is fundamentally different from signals like SIGTERM or SIGUSR1.